### DISCUSSION TITLE
How to use ipm for MIP root relaxation?

### DISCUSSION THREAD

**Discussion by tosttost** [ERGO-Code/HiGHS#861] [Ideas]
There is an option "solver", but if it is set to "ipm" the model is solved as LP not as MIP...

Motivation: we often have models where the root relaxation is very challenging for simplex algorithms (even commercial implementations). Therefore it is often crucial to solve the initial LP with barrier/interior point + crossover. Of course all subsequent LP are better solved by simplex.

**Comment by jajhall** [comment]
The behaviour that you observe is deliberate, and not something we'd want to change. It's necessary for benchmarking to force MPS files for MIPs to be solved as LPs.

We have plans to introduce concurrent solution of the root node in MIPs using barrier and dual simplex, and this will give you the performance that you require.

We could add an option to force the root node to be solved with barrier rather than simplex. What's your view @lgottwald ?

**Comment by lgottwald** [comment]
Yes we can definitely add the option already. I know that it helps a lot for certain LPs. I'll add an option later.

**Comment by SanPen** [comment]
Hi, was this implemented already?

**Reply by jajhall** [reply]
No. I started writing an LP solver that runs serial simplex versions and serial IPM concurrently, expecting this to achieve the best root node performance. However, it's still WIP. I probably won't get back to this before the summer
